source/simulator/core.cc

```diff
@@ -1043,6 +1046,7 @@ namespace aspect
             {
               coupling[x.velocities[d]][x.pressure] = DoFTools::always;
               coupling[x.pressure][x.velocities[d]] = DoFTools::always;
+              coupling[x.pressure][x.pressure] = DoFTools::always;
```

Comment:
I agree that the total pressure should include a contribution from gravitational potential, but the *dynamic* pressure does not. In this benchmark problem, we turned off gravity so the total pressure is the dynamic pressure. It is a purely kinematic flow benchmark in which buoyancy forces do not matter.
I think that if the discontinuity was represented correctly by the finite element, the equations should just yield the correct pressure, but in our case, there is a step discontinuity in velocity that couples into the pressure field because the laplacian of the velocity field is nonzero in the finite element approximation. I've produced as simple an example as I can that demonstrates this effect. Note that I am using initial global refinement level 1. For any higher refinement level, I see convergence problems. Note that I only prescribe the velocity in the upper left quadrant of the domain - I found that if velocity is prescribed everywhere I get division by zero errors during the stokes solve.
[pressure_for_wolfgang.prm.zip](https://github.com/geodynamics/aspect/files/1163171/pressure_for_wolfgang.prm.zip)



![image](https://user-images.githubusercontent.com/8154934/28430195-7a75ae7c-6d34-11e7-8044-39ecafb4bbc9.png)